Use JNA for TTY detection instead of JNR-POSIX

rnorth opened the pull request.

rnorth commented:
@bsideup this is the JNA version I mentioned; pretty simple at its core and has the benefit of bringing a smaller set of dependencies in. I've implemented this to be (hopefully) pretty resilient and fail-safe, so I'd hope that we can replace #6 with this and resolve #5. WDYT?

I've tested this for compatibility in OS X, Windows 10, and it's looking good in Travis.

bsideup commented:
@rnorth I tested it on my machine with `mvn test` (mvn 3.5.0) and the result was not colored o_O 
The same on Travis:
https://travis-ci.org/rnorth/visible-assertions/builds/300679495

bsideup commented:
@rnorth it fails with:
```
java.lang.ClassNotFoundException: org.codehaus.plexus.classworlds.launcher.Launcher
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
```

Perhaps the class was removed in Maven 3.5.0?

rnorth commented on src/main/java/org/rnorth/ansi/CapabilityDetection.java:
```diff
@@ -0,0 +1,46 @@
+            // Rough check for whether or not we're running in a Maven build
+            Class.forName("org.codehaus.plexus.classworlds.launcher.Launcher");
+            return true;
+        } catch (Exception ignored) {
```
`catch (Exception` is entirely wrong here, and same in `isTty` 😱 . Will fix.

bsideup replied:
`catch` is fine, at least here, it handles it correctly, the stacktrace I sent I got from adding `e.printStackTrace()` :)

rnorth commented:
Damn, this is why I shouldn't make last minute changes to a PR. I've updated the Maven detection code to something that still works on the latest version of Maven. Additionally, I've added Maven wrapper to avoid any potential build-time version discrepancies, since that was a risk.

rnorth commented:
Sigh, under Maven 3.5.2 we're not seeing coloured output on Travis, even after disabling batch mode. Maven detection is working correctly according to the logs, so I think Maven's copy of JAnsi is stripping the ANSI codes out.

I suspect I might have stumbled into [MNG-6220](https://issues.apache.org/jira/browse/MNG-6220) or similar; this is probably something not worth trying to resolve until the Maven situation is a bit more stable.

Given that this PR does improve local log colouring and also, more importantly, reduces scope for failures/conflicts, I'm going to go ahead and merge this.

rnorth commented:
I've requested Travis debug mode support for this repo, so when that materialises I'll see if there's anything we could do, but I suspect there won't be.
